### PR TITLE
Opt cache.add/addAll and importScripts out of a local service worker

### PR DESCRIPTION
**Pull request by jakearchibald**
Fixes https://github.com/whatwg/fetch/issues/303#issuecomment-265483054.

Related fetch PR https://github.com/whatwg/fetch/pull/435.

### CODE REVIEW

**Comment by jakearchibald** [comment]
Once this lands I still need to expose service worker clients to fetch events.

File: docs/index.bs [hunk 1]
```diff
@@ -1862,6 +1862,7 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
             1. If |r|'s [=request/url=]'s [=url/scheme=] is not one of "<code>http</code>" and "<code>https</code>", then:
                 1. [=fetch/Terminate=] all the ongoing <a>fetches</a> initiated by |requests| with reason *fatal*.
                 1. Break the loop.
+            1. If |r|'s [=request/client=]'s [=environment settings object/global object=] is a {{ServiceWorkerGlobalScope}} object, set |request|'s [=skip-local-service-worker flag=].
```
Comment (jungkees) [inline docs/index.bs]: Are you introducing a new flag here? Or did you intend to use _skip-service-worker flag_?

File: docs/index.bs [hunk 2]
```diff
@@ -2152,6 +2153,7 @@ spec: webappsec-referrer-policy; urlPrefix: https://w3c.github.io/webappsec-refe
         1. Let |serviceWorker| be |request|'s [=request/client=]'s [=environment settings object/global object=]'s [=ServiceWorkerGlobalScope/service worker=].
         1. If |serviceWorker|'s <a>imported scripts updated flag</a> is unset, then:
             1. Let |registration| be |serviceWorker|'s [=containing service worker registration=].
+            1. Set |request|'s [=skip-local-service-worker flag=].
```
Comment (jungkees) [inline docs/index.bs]: Same as above.

File: docs/index.bs (hunk 1 above)
Comment (jakearchibald) [inline docs/index.bs]: See https://github.com/whatwg/fetch/pull/435

**Comment by jungkees** [comment]
Also I think V1 should also take this behavior. Could you apply it to V1?

File: docs/index.bs (hunk 1 above)
Comment (jungkees) [inline docs/index.bs]: OK. Missed that. I'll check that.

**Comment by jakearchibald** [comment]
Note https://github.com/whatwg/fetch/pull/435#issuecomment-267262742 - skip-local-service-worker is going to change to become an enum.

**Comment by jakearchibald** [comment]
Yeah I'll add it to v1 too

**Comment by jakearchibald** [comment]
@jungkees The tests & fetch spec part of this have merged now. Happy for this to merge too?

**Pull request merged**